__tests__/workers/postUpdated.ts

```diff
@@ -593,3 +611,142 @@ it('should not update already approved post', async () => {
   expect(submission.id).toEqual(uuid);
   expect(submission.status).toEqual(SubmissionStatus.Accepted);
 });
+
+describe('on post create', () => {
+  beforeEach(async () => {
+    await saveFixtures(con, Source, [
+      {
+        id: COMMUNITY_PICKS_SOURCE,
+        name: 'Community recommendations',
+        image: 'sample.image.com',
+      },
+    ]);
+
+    await createDefaultUser();
+  });
+
+  describe('when data includes questions', () => {
+    it('creates a question record each one', async () => {
+      const uuid = randomUUID();
+      await createDefaultSubmission(uuid);
+
+      // pre-check
+      const questionsBefore = await con.getRepository(PostQuestion).find();
+      expect(questionsBefore.length).toEqual(0);
+
+      await expectSuccessfulBackground(worker, {
+        id: 'f99a445f-e2fb-48e8-959c-e02a17f5e816',
+        title: 'With questions',
+        url: `https://post.com/${uuid}`,
+        source_id: COMMUNITY_PICKS_SOURCE,
+        submission_id: uuid,
+        extra: {
+          questions: DEFAULT_QUESTIONS,
+        },
+      });
+
+      const post = await con
+        .getRepository(Post)
+        .findOneBy({ title: 'With questions' });
```

Comment:
I don't think you need the actual post here, should be able to match if you pass a already mocked post basically.
It would avoid time on database read-writes